src/components v2/Atoms/Forms/ErrorMessage/ErrorMessage.vue

```diff
@@ -0,0 +1,109 @@
+<template>
+  <div class="input-error" :class="{ hidden: !error }">
+    <p-icon class="icon" name="Error" :type="iconSize" />
+    <div class="text" :class="[size]">
+      {{ error }}
+    </div>
+  </div>
+</template>
+
+<script lang="ts">
+import PIcon from '@/components/Icon';
+import Vue, { PropType } from 'vue';
+
+const iconSizeMatrix = {
+  sm: 'xs',
+  md: 'sm',
+};
+
+export default Vue.extend({
+  name: 'PInputError',
+  components: {
+    PIcon,
+  },
+  props: {
+    error: {
+      type: String as PropType<string>,
+      default: '',
+    },
+    size: {
+      type: String as PropType<string>,
+      default: 'sm',
+    },
+  },
+  computed: {
+    iconSize(): string {
+      return iconSizeMatrix[this.size];
+    },
+  },
+});
+</script>
+
+<style lang="scss" scoped>
+.input-error {
+  color: var(--form-control-items-error-default-text-color);
+  display: flex;
+  align-items: center;
+  transform: translate(0, 0);
+  transition: 0.3s all;
+  opacity: 1;
+  min-height: 22px;
+  margin-top: 10px;
+
+  &.hidden {
+    opacity: 0;
+    transform: translate(0, -20px);
+  }
+
+  .icon {
+    margin-right: 4px;
+  }
+
+  .text {
+    font-family: var(--form-control-items-error-default-text-font-family);
+    font-weight: var(--form-control-items-error-default-text-font-weight);
+
+    &.sm {
+      font-size: calc(
+        var(--form-control-items-error-sm-text-typography-fontSize) * 1px
```

Comment:
I'm a bit average at naming things, but looking at some of the var names its a bit hard to tell the structure of the naming convention. Maybe more of a 'lets think/talk' about it going fwd rather than a 'lets change it now' kinda thing. They also a bit long, again I've not dedicated any time with the var names so maybe fits well in the overall approach. In this one ^ , I might have thought the `form-control-items` part of the name isn't super necessary , but maybe it is...just thinking out loud 👍

Reply:
I tend to agree, but `label` is too generic and can be misleading, so maybe we find a middle ground? `fc-label`?

This component is part of the organism `Form Control` which is a wrapper element within which we have all our main components like input text, radio buttons etc, many of which themselves have <label> elements within them.

Reply:
Unfortunately this is how we are currently flattening the CSS structure. THis is using the figma token structure.

I want to move towards `v-bind` in vue SFC from vue 3.2+ but that means we need to migrate.